app/Http/Controllers/Admin/HomeController.php

```diff
@@ -3,12 +3,26 @@
 namespace App\Http\Controllers\Admin;
 
 use App\Http\Controllers\Controller;
-use Illuminate\Http\Request;
+use App\Repositories\Book\BookRepositoryInterface;
```

Comment:
tại sao ko dùng Repository mà phải dùng interface?

Reply:
interface như một khuôn mẫu để cho các repository ở trên implement, trong 1 repo nó có nhiều các function, sử dụng interface nó bao quát hơn =))

Reply:
1. Nếu dùng repo, code có chạy không?
2. Ngoài việc interface là khuôn mẫu, có còn có tác dụng gì khác k?

Reply:
dùng repo code vẫn chạy, còn tác dụng khác em đang  tìm hiểu =))

Reply:
Khi DB thay đổi (ví dụ không dùng MySQL nữa), em sẽ phải viết lại repo (vì cách query khác). Khi đó, nếu em dùng repo ở đây, em sẽ phải sửa lại toàn bộ controller, còn nếu dùng interface, em chỉ cần sửa lại ở AppServiceProvider.